integration/test/src/org/broadinstitute/reaper/server/IntegrationContainersTest.scala

```diff
@@ -0,0 +1,150 @@
+
+class IntegrationContainersTest
+    extends TestKit(ActorSystem("reaper-int-test"))
+    with AsyncFlatSpecLike
+    with ForAllTestContainer
+    with ModelAutoDerivation
+    with Matchers {
+  lazy implicit val m: Materializer = ActorMaterializer()
+
+  lazy implicit val description: Description =
+    Description.createSuiteDescription(IntegrationContainersTest.super.getClass)
+
+  private val config: ReaperConfig = ReaperConfig
+    .load()
+    .fold(
+      failures => throw new RuntimeException(s"Could not load Reaper config: $failures"),
+      config => config
+    )
+
+  /** URL of vault server to use when getting bearer tokens for service accounts. */
+  private val vaultUrl = "https://clotho.broadinstitute.org:8200/"
+
+  /** Path in vault to the service account JSON to use in testing. */
+  private val vaultPath = "secret/dsde/gotc/test/clio/clio-account.json"
+
+  /** List of possible token-file locations, in order of preference. */
+  private val vaultTokenPaths = Seq(
+    File("/etc/vault-token-dsde"),
+    File(System.getProperty("user.home"), ".vault-token")
+  )
+
+  /** Pull service account credentials to use when accessing cloud resources from Vault. */
+  protected def loadCredentials(): GoogleCredentials = {
+    val vaultToken: String = vaultTokenPaths
+      .find(_.exists)
+      .map(_.contentAsString.stripLineEnd)
+      .getOrElse(sys.error("Vault token not found on filesystem!"))
+
+    val vaultConfig = new VaultConfig()
+      .address(vaultUrl)
+      .token(vaultToken)
+      .build()
+
+    val vaultDriver = new Vault(vaultConfig)
+    val accountJSON =
+      vaultDriver
+        .logical()
+        .read(vaultPath)
+        .getData
+        .asScala
+        .toMap[String, String]
+        .asJson
+
+    val jsonStream = new ByteArrayInputStream(
+      accountJSON.pretty(Printer.noSpaces.copy(dropNullValues = true)).getBytes()
+    )
+    ServiceAccountCredentials.fromStream(jsonStream)
+  }
+
+  private val reaperContainer = ReaperContainer.container
+  private val cromwellContainer = CromwellContainer.container
+  private val clioContainer: ClioDockerComposeContainer =
+    ClioDockerComposeContainer.waitForReadyLog(File("/tmp"))
+  private val postgresContainer = PostgreSQLContainer()
+
+  private lazy val db = Database.forURL(
+    s"jdbc:postgresql://${postgresContainer.containerIpAddress}:${postgresContainer.mappedPort(5432)}/test",
+    driver = "org.postgresql.Driver",
+    user = "test",
+    password = "test"
+  )
+
+  private lazy val clioWebClient: ClioWebClient =
+    ClioWebClient(
+      clioCredentials,
+      clioContainer.clioHost,
+      clioContainer.clioPort,
+      useHttps = false
+    )
+
+  private lazy val clioCredentials = new ClioCredentials(loadCredentials())
+
+  "IntegrationContainersTest" should "be able to connect and query postgres db" in {
+    assert(
+      db.createSession().createStatement().execute(postgresContainer.testQueryString)
```

Comment:
Do we define this `testQueryString` somewhere? Does this test work right now?

Reply:
testQueryString is defined by the postgresContainer within testcontainers that we use.
https://github.com/testcontainers/testcontainers-java/blob/master/modules/postgresql/src/main/java/org/testcontainers/containers/PostgreSQLContainer.java#L79

 And yes the test does work.

Reply:
Ah, I was looking at testcontainers-scala, which has some more indirection/abstraction